activesupport/lib/active_support/test_case.rb

```diff
@@ -22,8 +22,24 @@
 module ActiveSupport
   class TestCase < ::Minitest::Test
     Assertion = Minitest::Assertion
+    
+    # Class variable to store the parallel worker ID
+    @@parallel_worker_id = nil
 
     class << self
+      # Returns the current parallel worker ID if tests are running in parallel,
+      # nil otherwise.
+      #
+      #   ActiveSupport::TestCase.parallel_worker_id # => 2
+      def parallel_worker_id
+        @@parallel_worker_id
+      end
+      
+      # Sets the parallel worker ID
+      def parallel_worker_id=(value)
```

Comment:
I'm not sure we want this API to be public (just the setter) because if someone sets this in their application weird things will happen and they'll think it's a bug in Rails. Only the framework should set this value. So this should have a `# :nodoc` instead of docs.

```suggestion
      def parallel_worker_id=(value) # :nodoc:
```